sdks/python/apache_beam/runners/common.py

```diff
@@ -402,13 +403,17 @@ def finish(self):
   def _reraise_augmented(self, exn):
     if getattr(exn, '_tagged_with_step', False) or not self.step_name:
       raise
-    args = exn.args
-    if args and isinstance(args[0], str):
-      args = (args[0] + " [while running '%s']" % self.step_name,) + args[1:]
-      # Poor man's exception chaining.
-      raise type(exn), args, sys.exc_info()[2]
-    else:
-      raise
+    step_annotation = " [while running '%s']" % self.step_name
+    original_traceback = sys.exc_info()[2]
+    try:
+      new_exn = type(exn)(exn.args[0] + step_annotation, *exn.args[1:])
```

Comment:
There are a whole bunch of ways this could raise (some of which were not caught before, e.g. the exception type may not actually take args in its constructor). Now if args[0] isn't a str (well, basestring) I drop down to the other clause due to the TypeError.